File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/security/ChangePasswordRequest.java
package com.mieai.qqbot.admin.security;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record ChangePasswordRequest(
        @NotBlank @Size(max = 4096) String currentPassword,
        @NotBlank @Size(min = 12, max = 4096) String newPassword) {
}

File: qqbot-module-platform-admin/src/main/kotlin/com/mieai/qqbot/admin/security/ChangePasswordRequest.kt
package com.mieai.qqbot.admin.security

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class ChangePasswordRequest(
    @field:NotBlank @field:Size(max = 4096) val currentPassword: String?,
    @field:NotBlank @field:Size(min = 12, max = 4096) val newPassword: String?,
)
